presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/AddIntermediateAggregations.java

```diff
@@ -98,7 +98,7 @@ public boolean isEnabled(Session session)
         Optional<PlanNode> rewrittenSource = recurseToPartial(lookup.resolve(aggregation.getSource()), lookup, idAllocator);
 
         if (!rewrittenSource.isPresent()) {
-            return Optional.empty();
+            return Result.empty();
```

Comment:
You could also have `emptyResult()` static method in the `Rule` interface, could be easier to use. (for consideration)

Reply:
+1

Reply:
Now you do not need to import anything, after your suggestion you would need. I personally do not have any preferences, both work for me. If it is something which could make you happy, I will be glad to address that.

Do you still want this?

Reply:
> Now you do not need to import anything, after your suggestion you would need.

I don't understand. If that was a static method in `Rule`, implementations of `Rule` wouldn't need to import that.

Reply:
Lets leave it as it is. I do not see a big difference between `Result.empty()` and `emptyResult()`.